### DISCUSSION TITLE
Is there any way to display Rust `impl` blocks in outline panel?

### DISCUSSION THREAD

**Discussion by glebpinigin** [zed-industries/zed#22283] [Feature Requests]
Currently outline block for rust code doesn't display `impl` blocks that "owns" method definitions.

In the image you can see that there is a structure and 3 functions defined in source file.

![image](https://github.com/user-attachments/assets/7ead7533-d86e-481b-ac3c-8dc48d4a4f12)

What this outline doesn't display is that
a) Functions are defined in structure implementation
b) Functions implement different traits
c) The order of implementations is a little bit messed up

Second image is the reference from VSCode where all three above things are displayed

![image](https://github.com/user-attachments/assets/6de4aafa-b7dc-46da-a2e9-452c09a72dff)

The question is: is it possible to configure Zed's outline to display any of this things?

It is unclear for me whether this details about `impl` blocks should come from tree-sitter or from lsp. I think if none of mentioned features are currently available, this information might be important for understanding possibility of these to be implemented. AFAIK implementing new features on top of interactions with LSP is more difficult.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
